ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -203,23 +218,82 @@ func (h *FaultHandler) CheckNetworkBlackHolePort() func(http.ResponseWriter, *ht
 		rwMu.RLock()
 		defer rwMu.RUnlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Return the correct status state
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully checked status for fault", logger.Fields{
-			field.RequestType: requestType,
-			field.Request:     request.ToString(),
-			field.Response:    responseBody.ToString(),
-		})
+		ctx := context.Background()
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, requestTimeoutDuration)
+		defer cancel()
+
+		var responseBody types.NetworkFaultInjectionResponse
+		var statusCode int
+		chainName := fmt.Sprintf("%s-%s-%s", *request.TrafficType, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+		running, cmdOutput, cmdErr := h.checkNetworkBlackHolePort(ctxWithTimeout, request, chainName, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+
+		// We've timed out trying to check if the black hole port fault injection is running
+		if err := ctx.Err(); err != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(requestTimedOutError)
+		} else if cmdErr != nil {
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(cmdOutput)
+		} else {
+			statusCode = http.StatusOK
+			if running {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("running")
+			} else {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("not-running")
+			}
+			logger.Info("Successfully checked status for fault", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Response:    responseBody.ToString(),
+			})
+		}
+
 		utils.WriteJSONResponse(
 			w,
-			http.StatusOK,
+			statusCode,
 			responseBody,
 			requestType,
 		)
 	}
 }
 
+// checkNetworkBlackHolePort will check if there's a running black hole port within the task network namespace based on the chain name and the passed in required request fields.
+// It does so by calling iptables linux utility tool.
+func (h *FaultHandler) checkNetworkBlackHolePort(ctx context.Context, request types.NetworkBlackholePortRequest, chain, netNs string) (bool, string, error) {
+	cmdString := fmt.Sprintf(iptablesChainExistCmd, chain, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
```

Comment:
nit : we can use aws.StringValue instead for `*request.Protocol`

Reply:
should also do this in `CheckNetworkBlackHolePort` as well. perhaps we should just pass in protocol and port value as a function parameter instead of passing in the entire request